Code/autopkglib/recipes/__init__.py

```diff
@@ -0,0 +1,762 @@
+#!/usr/local/autopkg/python
+#
+# Copyright 2023 Nick McSpadden
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import glob
+import json
+import os
+import pathlib
+import plistlib
+import pprint
+import sys
+from dataclasses import dataclass, field
+from typing import Any, Dict, List, Optional
```

Comment:
I know that `typing` is littered all around the code right now, but maybe AutoPkg 3.0 should be the start of the transition to generic types, considering v2.6+ was built using Python 3.10.

I'm not saying go through _all_ the code to refactor right now, but maybe start with some of the net-new stuff.

References:
- https://stackoverflow.com/a/62033243/2233241
- https://stackoverflow.com/a/77300462/2233241

Reply:
This should be fixed now! Thanks for the advice.